File: 基礎題庫/Contents/b860/b860.c
#include<stdio.h>

int main(void)
{
	int candy, w, temp;
	
	while(scanf("%d %d", &candy, &w) != EOF)
	{
		int k = 0;
		int count = 0;
		
		while(w > 0)
		{
			temp = candy / 12;
			
			if(w < temp)
				temp = w;
			
			if(temp)
			{
				candy -= 11 * temp;
				w -= temp;
				k += temp;
				count += temp;
			}
			
			candy += k;
			k = 0;
			
			if(candy < 12)
			{
				w -= 12 - candy;
				candy = 12;
			}
		}
		
		printf("%d\n", count);
	}
	
	return 0;
}

File: 基礎題庫/Contents/b860/b860.py
import sys

for temp in sys.stdin:
    candy, w = map(int, temp.split())

    k = count = 0
    while w > 0:
        t = candy // 12

        if w < t:
            t = w
        
        if t > 0:
            candy -= 11 * t
            w -= t
            k += t
            count += t
        
        candy += k
        k = 0

        if candy < 12:
            w -= 12 - candy
            candy = 12
    
    print(count)
